addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -281,6 +281,10 @@ def test_03_sale_stock_delivery_partial(self):
         wizard = Form(self.env[(res_dict.get('res_model'))].with_context(res_dict['context'])).save()
         wizard.process_cancel_backorder()
 
+        #Check Exception error is logged on SO
+        exception_activity = self.env['mail.activity'].search([('res_id', '=', self.so.id), ('activity_type_id', '=', 6), ('res_model_id', '=', 461)])
+        self.assertTrue(exception_activity, 'Sale Stock: No exception error logged on SO after cancelling backorder for partial delivery')
```

Comment:
You should never ever hardcode in id numbers like this in your test because the numbers may change between test runs. Also your test description should say what is the correct behavior that should occur, not the incorrect behavior that has occurred.

```suggestion
        activity = self.env['mail.activity'].search([('res_id', '=', self.so.id), ('res_model', '=', 'sale.order')])
        self.assertEqual(len(activity), 1, 'When no backorder is created for a partial delivery, a warning error should be logged in its origin SO')
```

Reply:
Ahh! Thank you for your guidance! I overlooked the fact the id numbers will be different between builds. I made the changes for this test and i decided to add in an extra filter to find the exception activity in the case there may be multiple activities for the SO or would it not be necessary since the test method wouldn't have any other activity besides the exception?

Reply:
> Ahh! Thank you for your guidance! I overlooked the fact the id numbers will be different between builds. I made the changes for this test and i decided to add in an extra filter to find the exception activity in the case there may be multiple activities for the SO or would it not be necessary since the test method wouldn't have any other activity besides the exception?

I believe the extra filter isn't necessary since there won't be any other activity. Also it's better in general to not filter based on a string display name (i.e. `display_name`) since those strings can easily be changed during an improvement PR in the future. Thanks for the multi-picking extra test by the way :+1: 

So just remove the filter and we'll be good to r+!

Reply:
I removed the filter. Thank you for reviewing and teaching me along the way!